### DISCUSSION TITLE
Unable to load cluster list - NGINX - SSL - uwsgi.sock

### DISCUSSION THREAD

**Discussion by stadox** [rackslab/Slurm-web#389] [Q&A]
Hello,

I would like to ask you for a help.

Nowadays I have established working slurm + slurm web over HTTP.

Then I switched to HTTPS and set NGINX to HTTPS . But now once I'm opening the web page I'm getting error message "Unable to load cluster list"

Below are some printscreens.

![web](https://github.com/user-attachments/assets/a0c6d496-8c8a-4215-851d-f2e76a814c57)

![log](https://github.com/user-attachments/assets/c726c844-9b22-4870-a7cc-fc77b011cbcf)

**agent.ini**

> [service]
cluster=localcluster
interface=127.0.0.1
[slurmrestd]
socket=/run/slurmrestd/slurmrestd.socket
[cache]
enabled=yes

**gateway.ini**

> [ui]
host=https://slurm.mstep/
[agents]
url=https://slurm.mstep/agent
[service]
interface=127.0.0.1

**NGINX slurm-web.conf**

> server {
   listen 80;
    listen [::]:80;
    server_name slurm.mstep;
    return 301 https://$host$request_uri;
    }

> server {
   listen 443 ssl;
    server_name slurm.mstep;
    ssl_certificate /etc/nginx/ssl/slurm.mstep.crt;
    ssl_certificate_key /etc/nginx/ssl/slurm.mstep.key;
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers HIGH:!aNULL:!MD5;
    ssl_prefer_server_ciphers on;
    location / {
        include uwsgi_params;
        uwsgi_pass unix:/run/slurm-web-gateway/uwsgi.sock;
    }
    location /agent/ {
        include uwsgi_params;
        rewrite ^/agent/(.*)$ /$1 break;
        uwsgi_pass unix:/run/slurm-web-agent/uwsgi.sock;
    }
}

Thank you very much for an advice

Regards,

S.

**Comment by rezib** [comment]
Hello @stadox, can you give the full stacktrace out of `journalctl -u slurm-web-gateway-uwsgi.service` please?

**Reply by stadox** [reply]
Hello,

yes sure.

I do have for testing now self signed certificate.
I have seen some issue "ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate (_ssl.c:1129)"

![tracert_log](https://github.com/user-attachments/assets/5b203903-1042-40c0-8ef5-e5275d4fa341)


Could that be the reason why the web page isn't loading?

The self signed certificate it's OK itself but I wanted to test it before providing SSL certificate signed from CA.

Thank you

Regards,
S.

**Reply by rezib** [reply]
Yes, this is clearly the root cause of your error. Self-signed certificates are not yet supported by Slurm-web as of today, this is in the roadmap with #254.

I also noted that this error is not properly managed, I just opened #391 to avoid misleading reporting of `AttributeError`.

**Reply by stadox** [reply]
Hello,

OK now it works.

I have asked for SSL certificate signed from CA and applied to NGINX.

Now it's OK.

Thank you

Regards,
S.
